### PR TITLE
Release v2.0.0 Added support for EC2 based file gateway and VPC endpoints

### PR DESCRIPTION
**Pull request by prabirsekhri**
-Added module for EC2 based file gateways
-Added support for Storage Gateway VPC endpoints
-Added support for using Gateway endpoints for S3
-Added examples for S3 NFS file gateway on EC2

### CODE REVIEW

File: .header.md [hunk 1]
```diff
@@ -152,6 +182,74 @@ module "nfs_share" {
 
 The examples also includes "aws_kms_key" resource block to create a KMS key. For production deployments, you should pass in a key policy that restricts the use of the key based on your access requirements. Refer to this [link](https://docs.aws.amazon.com/kms/latest/developerguide/key-policies.html) for information. 
 
+## Networking Considerations
+
+### Storage Gateway interface VPC Endpoint configuration for EC2 Gateway
+
+Terraform Storage Gateway module allows you to optionally create an interface VPC Endpoint for Storage Gateway by setting create_vpc_endpoint=true. You can use this connection to activate your gateway and configure it to transfer data to AWS storage services without communicating over the public internet
+
+Example with VPC endpoint configuration :
```
Comment (AdamTylerLynch) [inline .header.md:191]:
```suggestion
Example with VPC endpoint configuration: 
```

File: .header.md [hunk 2]
```diff
@@ -152,6 +182,74 @@ module "nfs_share" {
 
 The examples also includes "aws_kms_key" resource block to create a KMS key. For production deployments, you should pass in a key policy that restricts the use of the key based on your access requirements. Refer to this [link](https://docs.aws.amazon.com/kms/latest/developerguide/key-policies.html) for information. 
 
+## Networking Considerations
+
+### Storage Gateway interface VPC Endpoint configuration for EC2 Gateway
+
+Terraform Storage Gateway module allows you to optionally create an interface VPC Endpoint for Storage Gateway by setting create_vpc_endpoint=true. You can use this connection to activate your gateway and configure it to transfer data to AWS storage services without communicating over the public internet
```
Comment (AdamTylerLynch) [inline .header.md:189]:
```suggestion
The Storage Gateway module allows you to optionally create an interface VPC Endpoint for Storage Gateway by setting `create_vpc_endpoint = true`. You can use this endpoint to activate your gateway and to transfer data to AWS storage services without communicating over the public internet.
```

File: .header.md [hunk 3]
```diff
@@ -152,6 +182,74 @@ module "nfs_share" {
...
+  gateway_type                       = "FILE_S3"
+  create_vpc_endpoint                = true
+  create_vpc_endpoint_security_group = true #if false define vpc_endpoint_security_group_id 
+  vpc_id                             = module.vpc.vpc_id
+  vpc_endpoint_subnet_ids            = module.vpc.private_subnets
+  gateway_private_ip_address         = module.ec2_sgw.private_ip
+}
+```
+
+ A security group is also needed for the VPC Endpoint. In the above example, the module handles creation of the security group. However, you may use the vpc_endpoint_security_group_id variable to associate an existing Security group with the VPC endpoint. Please see this [documentation](https://docs.aws.amazon.com/filegateway/latest/files3/gateway-private-link.html) which shows the Security Group requirements for Storage Gateway VPC endpoint. In this module, the security groups are already pre-configured with the required rules with the private IP address of the storage gateway appliance. The configuration can be found in the file [sg.tf](modules/aws-sgw/sg.tf)  file.
```
Comment (AdamTylerLynch) [inline .header.md:209]:
```suggestion
 A security group is needed for the VPC Endpoint. In the above example, the module handles creation of the security group. However, you may use the vpc_endpoint_security_group_id variable to associate an existing Security group with the VPC endpoint. Please see this [documentation](https://docs.aws.amazon.com/filegateway/latest/files3/gateway-private-link.html) which shows the Security Group requirements for a Storage Gateway VPC endpoint. In this module, the security groups are already pre-configured with the required rules with the private IP address of the storage gateway. The configuration can be found in the file [sg.tf](modules/aws-sgw/sg.tf)  file. 
```

File: .header.md [hunk 4]
```diff
@@ -152,6 +182,74 @@ module "nfs_share" {
...
+```hcl
+resource "aws_vpc_endpoint" "s3" {
+  vpc_id          = module.vpc.vpc_id
+  service_name    = "com.amazonaws.${var.aws_region}.s3"
+  route_table_ids = module.vpc.private_route_table_ids
+} 
+```
+### Storage Gateway Security Group Configuration for EC2 Gateway
+
+You can optionally create the security group and the required rules required for your gateway appliance by setting create_security_group = true. You can also limit access to range of ingress CIDR blocks in your network from where you require access to the storage gateway by modifying ingress_cidr_blocks attributes as shown in the example below.
```
Comment (AdamTylerLynch) [inline .header.md:224]: Copy from above

**Review by AdamTylerLynch — CHANGES_REQUESTED** [review]
Changes requested

**Comment by prabirsekhri** [comment]
Closed PR as a new PR is created with squashed commits

File: .header.md [hunk 5]
```diff
@@ -84,12 +82,44 @@ module "sgw" {
...
+## Usage with Amazon EC2 File Gateway module
+
+- Link to the S3 NFS Storage Gateway example for Amazon EC2: [s3-nfs-filegateway-ec2](examples/s3-nfs-filegateway-ec2)
+
+
+### [EC2 Storage Gateway module](modules/ec2-sgw/)
+
+```hcl
+module "ec2_sgw" {
+  source     = "aws-ia/storagegateway/aws//modules/ec2-sgw"
```
[the remark this answers is not in the document]
Comment (prabirsekhri) [reply .header.md:98]: Yes this was intentional for customers to run the module in their example directory. Setting it without the '/' results in Terraform thinking that is a relative path. │ Terraform failed to determine your intended installation method for remote module package "aws-ia/storagegateway/aws/modules/aws-sgw".
│ 
│ If you intended this as a path relative to the current module, use "./aws-ia/storagegateway/aws/modules/aws-sgw" instead. The "./" prefix
│ indicates that the address is a relative filesystem path.
╵

**Pull request closed without merging**